test/e2e/e2e.go

```diff
@@ -248,3 +253,49 @@ func RunE2ETests(t *testing.T) {
 
 	ginkgo.RunSpecsWithDefaultAndCustomReporters(t, "Kubernetes e2e suite", r)
 }
+
+func podFromManifest(filename string) (*api.Pod, error) {
+	var pod api.Pod
+	framework.Logf("Parsing pod from %v", filename)
+	data := generated.ReadOrDie(filename)
+	json, err := utilyaml.ToJSON(data)
+	if err != nil {
+		return nil, err
+	}
+	if err := runtime.DecodeInto(api.Codecs.UniversalDecoder(), json, &pod); err != nil {
+		return nil, err
+	}
+	return &pod, nil
+}
+
+// Run a test container to try and contact the Kubernetes api-server from a pod, wait for it
+// to flip to Ready, log its output and delete it.
+func RunKubernetesServiceTestContainer(c clientset.Interface, ns string) {
```

Comment:
minor suggestion: since it seems like nothing outside `e2e.go` uses this, unexport it by renaming `runKubernetesServiceTestContainer`?